### ISSUE TITLE
Final Exam Prep (worth 30%)

### ISSUE THREAD

**Issue by sprince0031** [sprince0031/UL_Masters#118]
Cover the following topics during prep:

- [ ] Deep Q Networks (for Atari)
  - [ ] Algorithm
  - [ ] DQN Architecture
  - [ ] Loss function
  - [ ] Performance
  - [ ] Training step
  - [ ] Experience buffer
  - [ ] Minibatch
  - [ ] Code for discount_rewards()
  - [ ] What are problems with DQNs? What causes erratic learning?
  - [ ] Catastrophic forgetting & maximisation bias
    - [ ] Fixed Q targets
    - [ ] Double Q learning
- [ ] Reinforcement Learning
  - [x] On vs Off policy
  - [x] Update policies
    - [x] SARSA
    - [x] Q Learning
    - [x] Why is Q learning generally better than SARSA?
  - [ ] K-armed bandits
  - [ ] What are policy gradients? Is policy gradient on-policy?
  - [x] Why Q-learning generates an optimal path with a higher loss? 
  - [x] Why is Q learning off-policy?
  - [ ] How is reinforce algorithm different from Q learning?
  - [ ] Code for implementation of Windy Grid World
  - [x] What is V, Q?
  - [x] Dynamic Programming
  - [x] Monte Carlo
  - [x] Temporal Difference
- [ ] CNNs
  - [ ] Alexnet
  - [ ] LeNet-5
  - [ ] GoogleNet
  - [ ] VGGNet
  - [x] Resnet
  - [ ] Inception module
  - [ ] Sigmoid activation function & problem
  - [ ] ReLUs
  - [x] 1x1 filter
  - [ ] Convolution in MV, Kernels
  - [ ] Ways to reduce overfitting
  - [ ] Perceptron training rule
  - [ ] Gradient descent - the delta rule
  - [ ] Back propagation algorithm
  - [ ] Purpose of bias in perceptron
- [ ] General questions
  - [ ] PSSH
  - [ ] POE model
  - [ ] Definition of ML (Tom Mitchell)
  - [ ] Explainable AI
  - [ ] What are the main challenges of ML?
  - [ ] Issues in computer vision systems
  - [ ] Alpha GO, Fold, Chess
  
> [!important]
> Exam on 07/05/2024 at 09:00 AM at the [gym in PESS building](https://maps.app.goo.gl/mx7qes9HSYSqSBGr8)

**Comment by sprince0031** [comment]
Wrote the exam to an okay-ish extent. Not super satisfied with my performance. The first question was about our implementations in our assignments which was a bummer because I hadn't started assignment 3 yet (submitted just today) 😞
